targets/TARGET_STM/TARGET_STM32F2/device/stm32f2xx_hal_def.h

```diff
@@ -130,19 +130,19 @@ static inline  void atomic_set_u32(volatile uint32_t *ptr, uint32_t mask)
 {
 	uint32_t newValue;
 	do {
-		newValue = (uint32_t)__LDREXW(ptr) | mask;
+		newValue = (uint32_t)__LDREXW((volatile uint32_t *)ptr) | mask;
```

Comment:
@ARMmbed/team-st-mcd FYI

Reply:
I can't see what this change is doing. You're casting something to its own type.

Reply:
This fix was added to feature branch more then a year ago. With updates to master and ST hal updates it is now irrelevant :-(
@0xc0170  - Shall i revert this as well?

Can see some fixes on similar terms on master https://github.com/ARMmbed/mbed-os/commit/c5a54382563b5b985833d2591ead8996cf427509

Reply:
Looks like a bad conflict resolution of that then - there was a `long` versus `uint32_t` mismatch, but that's gone now